### PR TITLE
Internal - Inlay Drawer Widget - Common Component

### PR DESCRIPTION
**Pull request by adityadivami**
# TITLE

## Description
Summary of changes

## PR Type
- [ ] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [Jira issue #](fill-in.org)

## Screenshots
Screenshot 2023-01-09 at 7 49 15 PM
Screenshot 2023-01-09 at 7 49 39 PM
Screenshot 2023-01-09 at 7 49 52 PM

### CODE REVIEW

File: app/cdap/components/common/InlayDrawerWidget/InlayDrawerWidget.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { withInfo } from '@storybook/addon-info';
+import { withKnobs } from '@storybook/addon-knobs';
+import { storiesOf } from '@storybook/react';
+import InlayDrawerWidget from 'components/common/InlayDrawerWidget';
+import React, { useState } from 'react';
+import { IActionsOptions } from './Menu';
```
Comment (adityadivami) [inline app/cdap/components/common/InlayDrawerWidget/InlayDrawerWidget.stories.tsx]: full path imports plz

File: app/cdap/components/common/InlayDrawerWidget/Menu.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  label: string;
+  value: string;
+  clickHandler: () => void;
+}
+
+const StyledContainer = styled.div`
+  display: flex;
+`;
+
+const ActionButton = styled(Button)`
```
Comment (sivaRajanaDivami) [inline app/cdap/components/common/InlayDrawerWidget/Menu.tsx:37]: Lets arrange the styled-components in lexicographical order

File: app/cdap/components/common/InlayDrawerWidget/Menu.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    letter-spacing: 0.5px;
+    text-align: left;
+  }
+`;
+
+export default function MenuListComposition({
+  dropdownOptions,
+}: Record<string, IActionsOptions[]>) {
+  const [open, setOpen] = useState(false);
+  const anchorRef = useRef(null);
```
Comment (sivaRajanaDivami) [inline app/cdap/components/common/InlayDrawerWidget/Menu.tsx]: Don't we need a type here?

File: app/cdap/components/common/InlayDrawerWidget/Menu.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+  const handleToggle = () => {
+    setOpen((prev) => !prev);
+  };
+
+  const handleClose = (event) => {
```
Comment (sivaRajanaDivami) [inline app/cdap/components/common/InlayDrawerWidget/Menu.tsx]: please add type for event parameter

File: app/cdap/components/common/InlayDrawerWidget/Menu.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    if (anchorRef.current && anchorRef.current.contains(event.target)) {
+      return;
+    }
+
+    setOpen(false);
+  };
+
+  function handleListKeyDown(event) {
```
Comment (sivaRajanaDivami) [inline app/cdap/components/common/InlayDrawerWidget/Menu.tsx]: please add type for event parameter

File: app/cdap/components/common/InlayDrawerWidget/Menu.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+      <ActionButton
+        ref={anchorRef}
+        aria-controls={open ? 'menu-list-grow' : undefined}
+        aria-haspopup="true"
+        onClick={handleToggle}
+        endIcon={<ArrowDropDownIcon />}
+      >
+        {T.translate(`${PREFIX}.actionsButtonLabel`)}
+      </ActionButton>
+      <Popper open={open} anchorEl={anchorRef.current} role={undefined} transition disablePortal>
```
Comment (sivaRajanaDivami) [inline app/cdap/components/common/InlayDrawerWidget/Menu.tsx]: Do we need a role prop here even though it's value is undefined?

File: app/cdap/components/common/InlayDrawerWidget/index.tsx [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+  overflow-y: scroll;
+  border-left: 1px solid ${grey[300]};
+`;
+
+const LeftContainer = styled(Container)`
+  border-left: none;
+  border-right: 1px solid ${grey[300]};
+`;
+
+const HeaderStyle = styled.header`
```
Comment (sivaRajanaDivami) [inline app/cdap/components/common/InlayDrawerWidget/index.tsx]: Please arrange these styled components in lexicographical order

File: app/cdap/components/common/InlayDrawerWidget/Menu.tsx (hunk 2 above)
Comment (adityadivami) [reply app/cdap/components/common/InlayDrawerWidget/Menu.tsx:37]: arranged styled components in lexicographical order

File: app/cdap/components/common/InlayDrawerWidget/Menu.tsx (hunk 4 above)
Comment (adityadivami) [reply app/cdap/components/common/InlayDrawerWidget/Menu.tsx]: added types to event parameters

File: app/cdap/components/common/InlayDrawerWidget/Menu.tsx (hunk 6 above)
Comment (adityadivami) [reply app/cdap/components/common/InlayDrawerWidget/Menu.tsx]: removed the role prop

File: app/cdap/components/common/InlayDrawerWidget/Menu.tsx (hunk 5 above)
Comment (adityadivami) [reply app/cdap/components/common/InlayDrawerWidget/Menu.tsx]: done

File: app/cdap/components/common/InlayDrawerWidget/index.tsx (hunk 7 above)
Comment (adityadivami) [reply app/cdap/components/common/InlayDrawerWidget/index.tsx]: done

File: app/cdap/components/common/InlayDrawerWidget/Menu.tsx (hunk 3 above)
Comment (adityadivami) [reply app/cdap/components/common/InlayDrawerWidget/Menu.tsx]: added type to anchorRef

File: app/cdap/components/common/InlayDrawerWidget/InlayDrawerWidget.stories.tsx (hunk 1 above)
Comment (adityadivami) [reply app/cdap/components/common/InlayDrawerWidget/InlayDrawerWidget.stories.tsx]: added full path imports

**Pull request closed without merging**